packages/cozy-mespapiers-lib/src/components/ModelSteps/helpers.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Check if a file is already selected in the state of the FormDataProvider
+ * @param {object} formData - State of the FormDataProvider
+ * @param {number} stepIndex - Used to know if the file is already selected for this step (Some paper have two Scan steps)
+ * @param {File|Blob} currentFile - File or Blob object
+ * @returns
+ */
+export const isFileAlreadySelected = (formData, stepIndex, currentFile) => {
+  for (const data of formData.data) {
+    if (data.stepIndex === stepIndex) {
+      if (currentFile.constructor.name === 'File') {
+        return (
+          data.file.name === currentFile.name &&
+          data.file.lastModified === currentFile.lastModified
+        )
+      }
+    }
+    return false
```

Comment:
Shouldn't this `return` statement be outside of the `for` loop?

Reply:
Non, la logique ici n'a pas changée de l'ancienne implémentation. Si nous ne rentrons pas dans le cas du `stepIndex`, alors il ne faut pas aller plus loin.
Ce contrôle est utile pour les quelques cas (permis/CIN/..) où nous avons 2 étapes de scan